doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+However, you can also find the specific Kconfig options required for each functionality to enable them separately.
+
+Heap usage and Matter-related statistics
+----------------------------------------
+
+Requirements:
+
+.. parsed-literal::
+   :class: highlight
+
+    CONFIG_SHELL=y
+    CONFIG_CHIP_STATISTICS=y
+    CONFIG_CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT=y
+    CONFIG_CHIP_MALLOC_SYS_HEAP=y
+    CONFIG_SYS_HEAP_RUNTIME_STATS=y
+
+You can measure heap usage by monitoring peak usage.
+To do this, you should first reset the current peak measurement, then read the peak usage, perform some operations on the device, and read the peak usage again.
+he difference between the initial and subsequent peak values indicates the number of bytes dynamically allocated by all operations performed by the device during that interval.
+This method allows you to check for memory leaks or determine the amount of memory specific operations dynamically allocate.
+By using these results, you can adjust the maximum heap size for your application, thereby optimizing RAM usage for other purposes.
+
+To measure the current peak heap usage, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter stat peak
+
+As a result you will get the following log:
+
+.. code-block:: console
+
+    Packet Buffers: 0
+    Timers: 2
+    TCP endpoints: 0
+    UDP endpoints: 1
+    Exchange contexts: 0
+    Unsolicited message handlers: 5
+    Platform events: 1
+    Heap allocated bytes: 208
+
+You can read the ``Heap allocated bytes`` value to learn about the current peak usage.
+In addition to heap statistics, you can also obtain other information about the Matter stack from this command.
+
+To clear the peak usage, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter stat reset
+
+NVS Settings storage usage
+--------------------------
+
+Requirements:
+
+.. parsed-literal::
+   :class: highlight
+
+    CONFIG_SHELL=y
+    CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL=y
+
+The NVS Settings usage may change during the device's lifetime.
+In Matter samples, if you set the ``storage_partition``size, you cannot change it using a firmware update.
+It is possible only by reflashing the device with new code that includes a new partition layout.
+Therefore, it is crucial to calculate and set the appropriate ``storage_partition`` size according to the project requirements, such as enabled endpoints, the cryptographic backend used, or the number of supported Matter fabrics.
+
+If this functionality is enabled, you can use :ref:`ug_matter_configuring_settings_shell`.
+
+To see the full list of available commands, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter_settings
+
+You will see a list of the available commands like this one:
```

Comment:
Are you sure about this? If I type it I get:

uart:~$ matter_settings
Please specify a subcommand.

Maybe it's because by default we use SHELL_MINIMAL.